src/vanilla/__tests__/DrawingEngine.test.ts

```
@@ -0,0 +1,927 @@
+    getSource: vi.fn((id: string) => sources[id]),
+    getLayer: vi.fn((id: string) => (layers.has(id) ? {} : undefined)),
+    queryRenderedFeatures: vi.fn(() => [] as unknown[]),
+    on: vi.fn((event: string, handler: (...args: unknown[]) => void) => {
+      if (!handlers[event]) handlers[event] = []
+      handlers[event].push(handler)
+    }),
+    off: vi.fn((event: string, handler: (...args: unknown[]) => void) => {
+      if (handlers[event]) handlers[event] = handlers[event].filter((h) => h !== handler)
+    }),
+    _trigger: (event: string, data: unknown) => {
+      handlers[event]?.forEach((h) => h(data))
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)